ui/components/app/confirm/scroll-to-bottom/scroll-to-bottom.tsx

```diff
@@ -0,0 +1,104 @@
+import { I18nContext } from '../../../../contexts/i18n';
+import {
+  Box,
+  ButtonIcon,
+  ButtonIconSize,
+  IconName,
+} from '../../../component-library';
+import type { StyleUtilityProps } from '../../../component-library/box';
+
+import {
+  BackgroundColor,
+  BlockSize,
+  Display,
+  FlexDirection,
+  IconColor,
+  BorderRadius,
+} from '../../../../helpers/constants/design-system';
+import { useScrollRequired } from '../../../../hooks/useScrollRequired';
+
+interface ContentProps extends StyleUtilityProps {
+  /**
+   * Elements that go in the page content section
+   */
+  children: React.ReactNode | React.ReactNode[];
+  /**
+   * Optional prop used to pass useScrollRequired's hasScrolledToBottom state up to the parent
+   */
+  hasScrolledToBottom?: boolean;
+  /**
+   * Optional setter function for hasScrolledToBottom
+   */
+  setHasScrolledToBottom?: Dispatch<SetStateAction<boolean>>;
+}
+
+const ScrollToBottom = ({
+  children,
+  hasScrolledToBottom: hasScrolledToBottomProp,
+  setHasScrolledToBottom: setHasScrolledToBottomProp,
+  ...props
+}: ContentProps) => {
+  const t = useContext(I18nContext);
+
+  const {
+    hasScrolledToBottom,
+    isScrollable,
+    isScrolledToBottom,
+    onScroll,
+    scrollToBottom,
+    ref,
+  } = useScrollRequired([]);
+
+  useEffect(() => {
+    if (hasScrolledToBottom && setHasScrolledToBottomProp) {
+      setHasScrolledToBottomProp(true);
+    }
+  }, [hasScrolledToBottom]);
+
+  return (
+    <Box
+      width={BlockSize.Full}
+      height={BlockSize.Full}
+      style={{
+        /** As a flex child, this ensures the element stretches the full available space without overflowing */
+        minHeight: '0',
+        /**
+         * This is for the scroll button. If we placed position: relative on the element below, with overflow: 'auto',
+         * the button would be positioned absolute to the entire content relative the scroll container. Thus, it would
+         * not stick to the bottom of the scroll container.
+         */
+        position: 'relative',
+      }}
+    >
+      <Box
+        display={Display.Flex}
+        flexDirection={FlexDirection.Column}
+        width={BlockSize.Full}
+        height={BlockSize.Full}
+        onScroll={onScroll}
+        ref={ref}
+        style={{ overflow: 'auto' }}
+        {...props}
+      >
+        {children}
+
+        {isScrollable && !isScrolledToBottom && (
+          <ButtonIcon
+            className={'confirm-scroll-to-bottom__button'}
+            onClick={scrollToBottom}
+            iconName={IconName.Arrow2Down}
+            ariaLabel={t('scrollDown')}
+            backgroundColor={BackgroundColor.backgroundDefault}
+            borderRadius={BorderRadius.full}
+            color={IconColor.primaryDefault}
+            display={Display.Flex}
+            size={ButtonIconSize.Md}
+          />
+        )}
+      </Box>
+    </Box>
+  );
+};
+
+export default ScrollToBottom;
```

Comment:
Hey @digiwand : is there a way for us to know that page has scrolled to the bottom, the plan is that for signature request pages sign button will be enabled only after user has scrolled and read whole message.

Reply:
Hi @jpuri, good question. 

there's commented code here: https://github.com/MetaMask/metamask-extension/pull/22349/files#diff-81455140e7e7ab4c7ec7834577a0b4f81e41361fabaf15737d8f67530ed6af07
```
  // TODO: use this to show/hide footer confirm button
  // const [isConfirmable, setIsConfirmable] = React.useState(false);
  // useEffect(() => {
  //   if (isConfirmable || !hasScrolledToBottom) {
  //     return;
  //   }
  //   setIsConfirmable(true);
  // }, [hasScrolledToBottom]);
```

maybe there is a more elegant solution. The idea here is that hasScrolledToBottom is, as far as our use-case goes, a value needed from the parent. The state and its set function may be passed to the ScrollToBottom component. ScrollToBottom component itself doesn't need the value, only sets it if available

Reply:
updated [1b5a61e](https://github.com/MetaMask/metamask-extension/pull/22349/commits/1b5a61ed916c307ebef7d31f777461cd33303520)